uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Wrapper/LeaveRequestDates.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+class CRM_HRLeaveAndAbsences_API_Wrapper_LeaveRequestDates implements API_Wrapper {
+
+  /**
+   * Adds Hours to the from_date and to_date fields, in case they're present but
+   * don't have that information.
+   *
+   * For the from_date field, the hour will be set to 00:00:00. For the to_date
+   * field, it will be set to 23:59:59
+   *
+   * @param array $apiRequest
+   *
+   * @return array
+   *   modified $apiRequest
+   */
+  public function fromApiInput($apiRequest) {
+    if($this->canHandleTheRequest($apiRequest)) {
+      $this->prepareDates($apiRequest);
+    }
+
+    return $apiRequest;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function toApiOutput($apiRequest, $result) {
+    return $result;
+  }
+
+  private function canHandleTheRequest($apiRequest) {
+    return $apiRequest['entity'] === 'LeaveRequest' &&
+           ($apiRequest['action'] === 'get' || $apiRequest['action'] === 'getfull');
+  }
+
+  /**
+   * If the query contains values for the from_date and to_date fields, this
+   * method makes sure they will always contain information about the hour.
+   *
+   * For the from_date field, it will set the hour as 00:00:00 if it's not
+   * present. For the to_date field, it will set it as 23:59:59 if it's not
+   * present.
+   *
+   * These are datetime fields, and the reason for this is that if there's a
+   * query like "to_date <= 2018-01-09" it will include all the leave requests
+   * ending on that date, regardless of the hour. Without this, such a query
+   * would be interpreted as "to_date <= 2018-01-09 00:00:00" by the database,
+   * and most of the requests ending on 2018-01-09 would not be included in the
+   * result.
+   *
+   * @param array $apiRequest
+   */
+  private function prepareDates(&$apiRequest) {
+    $params = &$apiRequest['params'];
+
+    $fromDate = $this->getValueFromParams($params, 'from_date');
+
+    if($fromDate && $this->dateIsValid($fromDate) && $this->dateIsMissingHours($fromDate)) {
+      $fromDate = new DateTime($fromDate);
+      $this->setValueOfParam($params,'from_date', $fromDate->format('Y-m-d 00:00:00'));
+    }
+
+    $toDate = $this->getValueFromParams($params, 'to_date');
+
+    if($toDate && $this->dateIsValid($toDate) && $this->dateIsMissingHours($toDate)) {
+      $toDate = new DateTime($toDate);
+      $this->setValueOfParam($params, 'to_date', $toDate->format('Y-m-d 23:59:59'));
+    }
+  }
+
+  private function dateIsValid($date) {
```

Comment:
Missing docblock, also for functions returning booleans I think it's common to follow the format `is__`

Reply:
I think it's just a matter of what reads better. If `$date` was an object, then I think that an `isValid()` method would be better, since we would write something like `if $date->isValid()`. That is, we are asking the date if it is valid. Here, however, we asking someone else to check if the date is valid, so I think this `if $this->dateIsValid($date)` reads better. But I don't mind changing if you think that the `is_` version is better

Reply:
Nah, it's fine. It's just as readable both ways